gui-agent/vmside.c

```diff
@@ -1679,19 +1692,43 @@ static void handle_button(Ghandles * g, XID winid)
     if (g->log_level > 1)
         fprintf(stderr,
                 "send buttonevent, win 0x%x type=%d button=%d\n",
-                (int) winid, key.type, key.button);
-    feed_xdriver(g, 'B', key.button, key.type == ButtonPress ? 1 : 0);
+                (int) winid, msg.type, msg.button);
+
+    bool is_button_press = msg.type == ButtonPress;
+
+    // click first, or some weird race condition may happen
+    // some applications close all context menus when focused, and the click will happen after
+    feed_xdriver(g, 'B', msg.button, is_button_press ? 1 : 0);
+
+    // Fake a "focus in" when mouse down on unfocused window.
+    if (is_button_press) {
+        // int _return_to;
+        // XID focused_winid;
+        // XGetInputFocus(g->display, &focused_winid, &_return_to);
+        // bool a = focused_winid != winid
+        // TODO: check if window is "top-level" (has border), and only focus if it or its child was not focused
+        // some applications get confused if you give focus to its context menu
+        // I'm not sure how this works yet
+        bool need_focus = false; 
```

Comment:
If anyone know how to do this properly, please send pull request to my branch (`fix-focus-2`)

Reply:
Windows have decorations if and only if they are not override-redirect.  Context menus can just be not given focus.